components/inputs/input-number.js

```diff
@@ -99,42 +92,84 @@ class InputNumber extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(Lit
 		this.maxExclusive = false;
 		this.minExclusive = false;
 		this.required = false;
-		this.trailingZeroes = false;
 
 		this._formattedValue = '';
 		this._inputId = getUniqueId();
-		this._numDecimalDigits = 0;
+		this._trailingZeroes = false;
+		this._valueTrailingZeroes = '';
+	}
+
+	get maxFractionDigits() {
+		return this._maxFractionDigits ?? Math.max(this.minFractionDigits, 3);
+	}
+	set maxFractionDigits(val) {
+		if (isNaN(val) || val < 0 || val > 20 || val < this.minFractionDigits) {
+			throw new RangeError('maxFractionDigits must be between 0 and 20 and >= minFractionDigits');
+		}
+		this._maxFractionDigits = val;
+		this._updateFormattedValue();
 	}
 
-	get value() { return this._value; }
+	get minFractionDigits() {
+		return this._minFractionDigits ?? 0;
+	}
+	set minFractionDigits(val) {
+		if (isNaN(val) || val < 0 || val > 20 || val < this.minFractionDigits) {
+			throw new RangeError('minFractionDigits must be between 0 and 20 and <= maxFractionDigits');
+		}
+		this._minFractionDigits = val;
+		this._updateFormattedValue();
+	}
+
+	get trailingZeroes() { return this._trailingZeroes; }
+	set trailingZeroes(val) {
+		this._trailingZeroes = val;
+		this._updateFormattedValue();
+	}
+
+	get value() {
+		if (this._value === undefined) return undefined;
+		const precision = Math.pow(10, this.maxFractionDigits);
+		return Math.round((this._value + Number.EPSILON) * precision) / precision;
+	}
 	set value(val) {
 		const oldValue = this.value;
-		try {
-			if (val === null) {
-				this._formattedValue = '';
-				this._value = undefined;
-			} else {
-				this._formattedValue = formatValue(
-					val,
-					this.minFractionDigits,
-					this.maxFractionDigits,
-					this.trailingZeroes,
-					this._numDecimalDigits
-				);
-				this._value = parseNumber(this._formattedValue);
-			}
-		} catch (err) {
-			this._formattedValue = '';
-			this._value = undefined;
```

Comment:
This used to take a `Number`, format it to a string and then parse that. That seemed overly complicated so I simplified it, which has the side effect that `elem.value = '80';` no longer works -- which was a test case in the `input-percent` tests. I think that should be OK though... setting a `Number` property to a string and having it work is bad IMO.